### PR TITLE
Add onBeforeChange trigger and options to item

### PR DESCRIPTION
**Pull request by damianfabian**
I just made this features for my own use and thought you would like to check if could be merged with your current library.

1.  Added onBeforeChange trigger that would be executed before the changeStep function is being called.
2. Added options to the steps API. Now, if you define the options in one step, you will be available to define the position of the navigation for that specific step.

### CODE REVIEW

**Comment by robophil** [comment]
Thanks @damianfabian, I'll look at this in a bit

**Comment by robophil** [comment]
Thanks for this @damianfabian, but did you get the example running after you made your changes?
I gave it a try, clicking next no longer works.

## steps to reproduce

1. `bower install`

1. `npm install`

1. `gulp serve`

**Comment by damianfabian** [comment]
I'm using it in my Tour right now, but i'll check again with the repository flow and let you know soon.

**Comment by damianfabian** [comment]
Sorry, I did not tried before with your repo, I made a little change, now should be fine for you. @Robophil

**Comment by robophil** [comment]
alright, let me give it another try

**Comment by robophil** [comment]
it works better now @damianfabian.
But when you get to the final item, and you want to navigate backward, it throws an error

```javascript
product-tour.js:183 Uncaught TypeError: Cannot read property 'element' of undefined
    at HTMLAnchorElement.<anonymous> (product-tour.js:183)
    at HTMLDivElement.dispatch (jquery.min.js:3)
    at HTMLDivElement.q.handle (jquery.min.js:3)
```

File: js/product-tour.js [hunk 1]
```diff
@@ -180,14 +180,14 @@ var ProductTour;
                 //go to prev step - if available
                 (!jQuery(event.target).hasClass('inactive')) &&
                 //call onBeforeChange trigger if specified
-                options.onBeforeChange(jQuery(global_items[getActiveStepCount()+1].element)) &&
+                options.onBeforeChange(jQuery(global_items[getActiveStepCount()+1].element)) ||
```
Comment (robophil) [inline js/product-tour.js]: This line would throw an error when you get to the last ltem

**Review by robophil — CHANGES_REQUESTED** [review]

File: js/product-tour.js (hunk 1 above)
Comment (damianfabian) [reply js/product-tour.js]: Yeah, I will try to save a time to fixed maybe later today.

**Comment by robophil** [comment]
Feel free to open this up again @damianfabian when you have time to fix this. Cheers!!

**Pull request closed without merging**